docs/attributes-registry/security-rule.md

```diff
@@ -0,0 +1,25 @@
+<!--- Hugo front matter used to generate the website version of this page:
+--->
+
+<!-- NOTE: THIS FILE IS AUTOGENERATED. DO NOT EDIT BY HAND. -->
+<!-- see templates/registry/markdown/attribute_namespace.md.j2 -->
+
+# Security Rule
+
+## Security Rule
```

Comment:
I know I'm a bit late here, I've been following this.

To some extent, it's unclear to me who will be generating this data, so if you can speak to how it's used in Elastic Common Schema and instances of logs that generate this data that could help.

Some major comments:

- `Rule` vs `Security Rule` vs. `Policy` - it's hard to understand what the name/scope should be here, as it's very broad.  E.g. the way this is phrased now, this could apply to [OPA](https://www.openpolicyagent.org/), which has a very clear definition of what is a policy and a rule (and I'd argue, a good foundation we can align with ECS as a way forward).  I'm not in the details, but I *think* what you have aligns with rule there, so `policy.rule` may be a good namespace here, not that I want to litigate all that right now.
- Namespacing and re-use within Semantic Conventions.  I'm afraid we haven't resolved the discussion of `embed` sufficiently AND I feel like if we did, there'd be a LOT less contention on this PR.
- The lack of an "Event", "span" or "Metric" defined for these makes it hard for general semconv maintainers (like myself) to understand what the eventual us case will be.  However, given where Event stands in semconv, that would also block this PR so I know you want to make progress.
- A couple nit-picky things, e.g. `license` being something that I don't understand yet (I'm not a domain expert here, so don't take this as a blocker, just something I think would need to be justified).

In the description, you call out several examples of who would use these attributes.  Do you have example events that are produced by those things we could look at or point to?  Until we unblock Events effectively and/or make progress on `embed` I think this could do a lot for "general" reviewers to be able to evaluate this.

TL;DR; I don't feel capable to approve this PR right now.  I suggest one of two options:

1. Help educate general semconv approvers on this topic, with more links, example instrumentation, etc.
2. Bring in security experts (e.g. OPA folks, network policy folks, etc.) who can comment on this PR or approve it.

Reply:
I can try to address some of these, @trisch-me might know more or have more feedback

The usual security use case of these rule fields would be in alert event logs. A security product would run on endpoints monitoring/protecting systems by evaluating sets of rules loaded into the product. When a rule evaluates to true, the endpoint generates an alert event that includes information on what rule triggered the alert. SIEM products can also run detection rules on the data they have already ingested, which is an important use case, but less relevant to open telemetry, since the data is staying within the backend.

Examples of security rules:
* Sigma - https://github.com/SigmaHQ/sigma
* Falco - https://github.com/falcosecurity/rules
* Elastic - https://github.com/elastic/detection-rules

The Falco documentation has some easy to understand documentation of what a security rule is: https://falco.org/docs/rules/basic-elements/

I think within security, the differences of rule vs. policy isn't always consistent or clear. Sometimes "rule" is only the detection condition, and sometimes the conditional and response action combined, so more like a OPA policy.  Falco rules define the detection condition and response, by defining the priority and output content of the resulting alert. Elastic and Sigma rules only define the detection condition, and the response action is defined separately.

I'm unsure if it's common for OPA rules, or other non-security rules to be published in the same way security rules are, so I don't know if things like version, license, etc apply to them. If they do, I agree this namespace could be made more broad to cover other use cases too. 

I could provide a raw alert event with ECS if you want, but it could be hard to understand and might not be too useful. Instead here's a screenshot of the alert UI, which is generated with the ECS event fields:
![Screenshot 2024-08-20 at 2 57 13 PM](https://github.com/user-attachments/assets/9f9c08b5-d56e-4202-972a-a6eb258562a9)
![Screenshot 2024-08-20 at 2 59 51 PM](https://github.com/user-attachments/assets/5107cae9-41d2-4312-9a52-ed6ec7992089)

For the license, the rules are usually published as code, so the license is just the normal code copyright license that applies to it.

Reply:
Regarding my concerns.

> Rule vs Security Rule vs. Policy

I think we sufficiently entertained this and it's ok to close.

> Namespacing and re-use within Semantic Conventions. I'm afraid we haven't resolved the discussion of embed sufficiently AND I feel like if we did, there'd be a LOT less contention on this PR.

I think we've had enough time to think through this and the PR is fine as-is regarding this.

> The lack of an "Event", "span" or "Metric" defined for these makes it hard for general semconv maintainers (like myself) to understand what the eventual us case will be. However, given where Event stands in semconv, that would also block this PR so I know you want to make progress.

While I think Event is unblocked and I'd like to avoid adding attribute conventions that do NOT include signals, I'm deffering to other @open-telemetry/specs-semconv-maintainers here.

> A couple nit-picky things, e.g. license being something that I don't understand yet (I'm not a domain expert here, so don't take this as a blocker, just something I think would need to be justified).

These were sufficiently addressed.